src/components/ImagesCarousel.js

```diff
@@ -24,23 +24,30 @@ const TouchableImage = ({ navigation, item }) => {
       }
       activeOpacity={0.8}
     >
-      <Image source={item.picture} />
+      {children}
     </TouchableOpacity>
   );
 };
 
 TouchableImage.propTypes = {
   navigation: PropTypes.object.isRequired,
-  item: PropTypes.object.isRequired
+  item: PropTypes.object.isRequired,
+  children: PropTypes.object.isRequired
 };
 
-
 export const ImagesCarousel = ({ data, navigation, fetchPolicy }) => {
-  const { dimensions } = useContext(OrientationContext);
+  const { orientation, dimensions } = useContext(OrientationContext);
 
   const renderItem = ({ item }) => {
     const { routeName, params } = item.picture;
 
+    // special image width for carousel, to be not full width on landscape
+    const width = orientation === 'landscape' ? dimensions.height : dimensions.width;
+    const imageStyle = {
+      height: imageHeight(width),
+      width
```

Comment:
imageHeight from imageHelper , what is it happening here ? from comment image will not be full width on landscape , how is this happening ?

Reply:
imageStyle is set with the values passed.

height will be the height based on the width calculated in the helper, which i described in a comment before.
width will be the width depending on the orientation, which ensures, that the image has the width of the device in portrait and the height of the device in landscape. width in portrait equals height in landscape. so at eh end, images are always same sized on portrait and landscape, right?!